docfiles/docs.js

```diff
@@ -280,6 +281,135 @@ function renderSnippets() {
+    cardEl.appendChild(contentEl);
+
+    return cardEl;
+}
+
+function setupLangPicker() {
+    var appTheme = pxt.appTarget.appTheme;
+    var initialLang = pxt.Util.normalizeLanguageCode(getCookieLang())[0];
+    var modalContainer = document.querySelector("#langmodal");
+
+    if (appTheme.availableLocales && appTheme.selectLanguage) {
+        var localesContainer = document.querySelector("#availablelocales");
+        appTheme.availableLocales.forEach(function(locale) {
+            var card = languageOption(locale);
+            localesContainer.appendChild(card);
+        });
+
+        modalContainer.className += `  ${appTheme.availableLocales.length > 4 ? "large" : "small"}`
+
+        var disabledTabIndexAttribute = "langmodaldisabledtabindex";
+
+        $(modalContainer).modal({
+            onShow: function() {
+                const focusableElements = document.querySelectorAll('a, button, input, [tabindex]:not([tabindex="-1"])');
+                for (var i = 0; i < focusableElements.length; i++) {
+                    var el = focusableElements[i];
+                    if (!modalContainer.contains(el)) {
+                        el.setAttribute(disabledTabIndexAttribute, el.getAttribute("tabindex") || "0");
+                        el.setAttribute("tabindex", "-1");
+                    }
+                }
+            },
+            onHide: function(){
+                const returnFocusTo = document.querySelectorAll("[disabledtabindex]");
+                for (var i = 0; i < returnFocusTo.length; i++) {
+                    var el = returnFocusTo[i];
+                    el.setAttribute("tabindex", el.getAttribute(disabledTabIndexAttribute));
+                    el.setAttribute(disabledTabIndexAttribute, undefined);
+                }
+            }
+        });
+
+        var langPicker = document.querySelector("#langpicker");
+        langPicker.onclick = function() {
+            $(modalContainer).modal('show');
+        }
+        langPicker.onkeydown = handleEnterKey;
+
+        var closeIcon = document.querySelector(".closeIcon");
+        closeIcon.onclick = function() {
+            $(modalContainer).modal('hide');
+        }
+        closeIcon.onkeydown = handleEnterKey;
+
+        var buttons = modalContainer.querySelectorAll(".ui.button");
+        for (var i = 0; i < buttons.length; i++) {
+            buttons[i].onkeydown = handleEnterKey;
+        }
+
+        var langOptions = modalContainer.querySelectorAll(".langoption");
+
+        for (var i = 0; i < langOptions.length; i++) {
+            var currentOption = langOptions[i];
+
+            currentOption.onclick =  function(e) {
+                var langId = e.currentTarget.dataset.lang;
+                if (!pxt.Util.allLanguages[langId]) {
+                    return;
+                }
+                setCookieLang(langId);
+                if (langId !== initialLang) {
+                    pxt.tickEvent("menu.lang.changelang." + langId);
```

Comment:
Fix this in the new format. dynamic langId shouldn't be done here. 
        pxt.tickEvent(`menu.lang.changelang`, { lang: langId });

Reply:
Also probably add a notion of docs too.
pxt.tickEvent("menu.lang.changelang", { lang: langId, docs: "true" });